### ISSUE TITLE
Support "asValue" in InvalidValueTreatment

### ISSUE THREAD

**Issue by wf-r** [jpmml/sklearn2pmml#298]
Hi,

currently, "asValue" is not an allowed parameter for InvalidValueTreatment of a Domain, however one has to set it in order to use InvalidValueReplacement (which is a working parameter): http://dmg.org/pmml/v4-4/MiningSchema.html#xsdType_INVALID-VALUE-TREATMENT-METHOD

Would it be possible to allow "asValue"?

* Python: 3.8.6
* sklearn2pmml: 0.74.2

Best
Wolfgang

**Comment by vruusmann** [comment]
@wf-r Try re-installing this package from GitHub, and see if you're happy with the results.

Looks like the mechanism for specifying the invalid value replacement value was completely rearranged between PMML 4.3 and 4.4 schema versions. In fact, the PMML 4.3 schema version did not support it at all, and it was enabled as JPMML vendor extension (the `MiningField@x-invalidValueReplacement` attribute).

To summarize:
PMML 4.3 - `<MiningField invalidValueTreatment="asIs" x-invalidValueReplacement="0"/>`
PMML 4.4 - `<MiningField invalidValueTreatment="asValue" invalidValueReplacement="0"/>`

**Comment by wf-r** [comment]
Thank you for your quick reply. I re-installed from GitHub, however I get the following error:

```
Standard error:
Sep. 13, 2021 1:44:58 NACHM. org.jpmml.sklearn.Main run
INFO: Parsing PKL..
Sep. 13, 2021 1:44:58 NACHM. org.jpmml.sklearn.Main run
INFO: Parsed PKL in 17 ms.
Sep. 13, 2021 1:44:58 NACHM. org.jpmml.sklearn.Main run
INFO: Converting PKL to PMML..
Sep. 13, 2021 1:44:58 NACHM. org.jpmml.sklearn.Main run
SEVERE: Failed to convert PKL to PMML
java.lang.IllegalArgumentException: as_value
	at sklearn2pmml.decoration.DomainUtil.parseInvalidValueTreatment(DomainUtil.java:70)
	at sklearn2pmml.decoration.Domain.encodeFeatures(Domain.java:72)
	at sklearn2pmml.decoration.DiscreteDomain.encodeFeatures(DiscreteDomain.java:72)
	at sklearn.Transformer.encode(Transformer.java:70)
	at sklearn2pmml.decoration.MultiDomain.encodeFeatures(MultiDomain.java:51)
	at sklearn.Transformer.encode(Transformer.java:70)
	at sklearn_pandas.DataFrameMapper.initializeFeatures(DataFrameMapper.java:73)
	at sklearn.Initializer.encodeFeatures(Initializer.java:48)
	at sklearn.Transformer.encode(Transformer.java:70)
	at sklearn.Composite.encodeFeatures(Composite.java:119)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:211)
	at org.jpmml.sklearn.Main.run(Main.java:217)
	at org.jpmml.sklearn.Main.main(Main.java:143)

Exception in thread "main" java.lang.IllegalArgumentException: as_value
	at sklearn2pmml.decoration.DomainUtil.parseInvalidValueTreatment(DomainUtil.java:70)
	at sklearn2pmml.decoration.Domain.encodeFeatures(Domain.java:72)
	at sklearn2pmml.decoration.DiscreteDomain.encodeFeatures(DiscreteDomain.java:72)
	at sklearn.Transformer.encode(Transformer.java:70)
	at sklearn2pmml.decoration.MultiDomain.encodeFeatures(MultiDomain.java:51)
	at sklearn.Transformer.encode(Transformer.java:70)
	at sklearn_pandas.DataFrameMapper.initializeFeatures(DataFrameMapper.java:73)
	at sklearn.Initializer.encodeFeatures(Initializer.java:48)
	at sklearn.Transformer.encode(Transformer.java:70)
	at sklearn.Composite.encodeFeatures(Composite.java:119)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:211)
	at org.jpmml.sklearn.Main.run(Main.java:217)
	at org.jpmml.sklearn.Main.main(Main.java:143)
```

**Comment by vruusmann** [comment]
> java.lang.IllegalArgumentException: as_value
>	at sklearn2pmml.decoration.DomainUtil.parseInvalidValueTreatment(DomainUtil.java:70)

Yep - I totally forgot that this new enum constant also needs Java-side backing.

Reopening now. Will re-close when the JPMML-SkLearn library has been brought up to date.

**Comment by vruusmann** [comment]
> this still does not seem enough:
>
> java.lang.IllegalArgumentException
>   at org.jpmml.converter.InvalidValueDecorator.<init>(InvalidValueDecorator.java:36)
>

A totally valid complaint - the "asIs" enum constant has been thoroughly used in place of "asValue". Was not properly updated when migrating from PMML 4.3 to 4.4.

The JPMML-Evaluator library also appears to be affected.

**Comment by wf-r** [comment]
Thanks for the fix, I can confirm that the issue no longer exists.
